system/jlib/jsocket.cpp

```diff
@@ -6301,6 +6301,87 @@ inline bool appendv4range(SocketEndpointArray *array,char *str,SocketEndpoint &e
     return true;
 }
 
+bool SocketEndpointArray::fromName(const char *name, unsigned defport)
+{
+    // Lookup a single name that may resolve to multiple IPs in a headless service scenario
+    StringArray portSplit;
+    portSplit.appendList(name, ":");
+    switch (portSplit.ordinality())
+    {
+    case 2:
+        defport = atoi(portSplit.item(1));
+        name = portSplit.item(0);
```

Comment:
Are we changing name here for caller ?

Reply:
No, just using the bit before : in the rest of this function